src/Microsoft.CodeQuality.Analyzers/Core/QualityGuidelines/UseLiteralsWhereAppropriate.Fixer.cs

```diff
@@ -20,11 +26,91 @@ public sealed override FixAllProvider GetFixAllProvider()
             return WellKnownFixAllProviders.BatchFixer;
         }
 
-        public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
         {
-            // Fixer not yet implemented.
-            return Task.CompletedTask;
-            
+            SyntaxNode root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+            SyntaxNode declaration = root.FindNode(context.Span);
+            declaration = SyntaxGenerator.GetGenerator(context.Document).GetDeclaration(declaration, DeclarationKind.Field);
+            var fieldFeclaration = GetFieldDeclaration(declaration);
+            if (fieldFeclaration == null)
+            {
+                return;
+            }
+
+            // We cannot have multiple overlapping diagnostics of this id.
+            Diagnostic diagnostic = context.Diagnostics.Single();
+
+            context.RegisterCodeFix(
+                new MyCodeAction(
+                    MicrosoftQualityGuidelinesAnalyzersResources.UseLiteralsWhereAppropriateCodeActionTitle,
+                    cancellationToken => ToConstantDeclarationAsync(context.Document, fieldFeclaration, cancellationToken)),
+                diagnostic);
+        }
+
+        private async Task<Document> ToConstantDeclarationAsync(Document document, SyntaxNode fieldDeclaration, CancellationToken cancellationToken)
+        {
+            var editor = await DocumentEditor.CreateAsync(document, cancellationToken).ConfigureAwait(false);
+
+            SyntaxTriviaList leadingTrivia = new SyntaxTriviaList();
+            SyntaxTriviaList trailingTrivia = new SyntaxTriviaList();
+
+            SyntaxTokenList newModifiers = new SyntaxTokenList();
+            foreach (SyntaxToken modifier in GetModifiers(fieldDeclaration))
+            {
+                if (IsStaticKeyword(modifier) || IsReadonlyKeyword(modifier))
+                {
+                    // The associated analyzer ensures we'll only get in the fixer if both 'static' and 'readonly'
+                    // keywords are in the declaration. Because their order is not relevant, we detect if both
+                    // have been passed by inspecting whether leading and trailing trivia are non-empty. 
+                    if (leadingTrivia.Count == 0 && trailingTrivia.Count == 0)
```

Comment:
That results into:
```
public /*trivia3*/     /*trivia1*/ const /*trivia2*/  /*trivia4*/ string f1 = """";
```

So the order of comments changes. I agree it's not the best, but at least we don't throw away information.

Thanks for merging. Perhaps if you can think of a better algorithm we can create a separate issue for this....